plyer/platforms/android/smsreceive.py

```diff
@@ -14,43 +12,48 @@
 Intent = autoclass('android.content.Intent')
 Bundle = autoclass('android.os.Bundle')
 GenericBroadcastReceiver = autoclass(
-    'org.kivy.android.GenericBroadcastReceiver'
-)
+    'org.kivy.android.GenericBroadcastReceiver')
 SmsMessage = autoclass('android.telephony.SmsMessage')
 Toast = autoclass('android.widget.Toast')
 
 
 class BroadcastReceiver(PythonJavaClass):
     __javainterfaces__ = [
         'org/kivy/android/GenericBroadcastReceiverCallback']
+
     __javacontext__ = 'app'
 
     def __init__(self, *args, **kwargs):
-        PythonJavaClass.__init__(self, *args, **kwargs)
+        super(BroadcastReceiver, self).__init__(self, *args, **kwargs)
+        print "reached Broadcast Receiver class"
 
     @java_method(
         '(Landroid/content/Context;Landroid/content/Intent;)V')
     def onReceive(self, context, intent):
-        bundle = intent.getExtras()
-        messages = None
-        string = ""
-        if not bundle:
-            raise ReceiveError()
-        else:
-            pdus = bundle.get('pdus')
-            messages = SmsMessage[len(pdus)]
-            for i in range(len(messages)):
-                messages[i] = SmsMessage.createFromPdu(
-                    list(bytearray(pdus[i])))
-                string += "Message from " +\
-                    messages[i].getOriginatingAddress() +\
-                    " :" + str(messages[i].getMessageBody()) + "\n"
-            Toast.makeText(context, string, Toast.LENGTH_SHORT).show()
-
+        print "reached onReceive method"
+        try:
+            bundle = intent.getExtras()
+            messages = None
+            string = ""
+            if not bundle:
+                raise ReceiveError()
+            else:
+                pdus = bundle.get('pdus')
+                messages = SmsMessage[len(pdus)]
+                for i in range(len(messages)):
+                    messages[i] = SmsMessage.createFromPdu(
+                        list(bytearray(pdus[i])))
+                    string += "Message from " +\
+                        messages[i].getOriginatingAddress() +\
+                        " :" + str(messages[i].getMessageBody()) + "\n"
+                Toast.makeText(context, string, Toast.LENGTH_SHORT).show()
+        except:
+            import traceback
```

Comment:
Sorry, but what is this line actually good for? The only thing what will happen if there's any error is that the app won't crash because of that. Rather than this try to implement some `on_error` method or something, if you want to go this way :)

Also, if some try-except code is required, try to wrap it only around a tiny bit of code with an expected error handled, e.g. `except ImportError:`

Exactly because of this change your `raise RecieveError()` will fail and only because of the traceback print will be visible (hopefully).